imblearn/under_sampling/edited_nearest_neighbours.py

```diff
@@ -382,3 +382,171 @@ def _sample(self, X, y):
+    Original dataset shape Counter({1: 500, -1: 268})
+    >>> allknn = AllKNN(random_state=42)
+    >>> X_res, y_res = allknn.fit_sample(X, y)
+    >>> print('Resampled dataset shape {}'.format(Counter(y_res)))
+    Resampled dataset shape Counter({-1: 268, 1: 249})
+
+    References
+    ----------
+    .. [1] I. Tomek, "An Experiment with the Edited Nearest-Neighbor
+       Rule," IEEE Transactions on Systems, Man, and Cybernetics, vol. 6(6),
+       pp. 448-452, June 1976.
+
+    """
+
+    def __init__(self, return_indices=False, random_state=None,
+                 size_ngh=3, kind_sel='all', n_jobs=-1):
+        super(AllKNN, self).__init__()
+        self.return_indices = return_indices
+        self.random_state = random_state
+        self.size_ngh = size_ngh
+        self.kind_sel = kind_sel
+        self.n_jobs = n_jobs
+        self.enn_ = EditedNearestNeighbours(
+            return_indices=self.return_indices,
+            random_state=self.random_state,
+            size_ngh=self.size_ngh,
+            kind_sel=self.kind_sel,
+            n_jobs=self.n_jobs)
+
+    def fit(self, X, y):
+        """Find the classes statistics before to perform sampling.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Matrix containing the data which have to be sampled.
+
+        y : ndarray, shape (n_samples, )
+            Corresponding label for each sample in X.
+
+        Returns
+        -------
+        self : object,
+            Return self.
+
+        """
+        super(AllKNN, self).fit(X, y)
+        self.enn_.fit(X, y)
+
+        return self
+
+    def _sample(self, X, y):
+        """Resample the dataset.
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_samples, n_features)
+            Matrix containing the data which have to be sampled.
+
+        y : ndarray, shape (n_samples, )
+            Corresponding label for each sample in X.
+
+        Returns
+        -------
+        X_resampled : ndarray, shape (n_samples_new, n_features)
+            The array containing the resampled data.
+
+        y_resampled : ndarray, shape (n_samples_new)
+            The corresponding label of `X_resampled`
+
+        idx_under : ndarray, shape (n_samples, )
+            If `return_indices` is `True`, a boolean array will be returned
+            containing the which samples have been selected.
+
+        """
+
+        if self.kind_sel not in SEL_KIND:
+            raise NotImplementedError
+
+        X_, y_ = X.copy(), y.copy()
+
+        if self.return_indices:
+            idx_under = np.arange(X.shape[0], dtype=int)
+
+        prev_len = y.shape[0]
+
+        for curr_size_ngh in range(1, self.size_ngh + 1):
+            self.logger.debug('Apply ENN size_ngh #%s', curr_size_ngh)
+            # updating ENN size_ngh
+            self.enn_.size_ngh = curr_size_ngh
+            if self.return_indices:
+                X_, y_, idx_ = self.enn_.fit_sample(X_, y_)
```

Comment:
I am not sure that the indexes are correct. The indexes will be taken from `X_` iteratively which after several iteration are not corresponding to the indexes of `X`.

I think that at the first iteration `idx_` is correct then you need to pick up `idx_[idx_]`, so you will need a temporary variable.